verisignmdns/src/main/java/denominator/verisignmdns/VerisignMDNSAllProfileResourceRecordSetApi.java

```diff
@@ -0,0 +1,127 @@
+package denominator.verisignmdns;
+
+import static denominator.common.Preconditions.checkArgument;
+import static denominator.common.Preconditions.checkNotNull;
+import static denominator.model.ResourceRecordSets.notNull;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import denominator.AllProfileResourceRecordSetApi;
+import denominator.Credentials;
+import denominator.common.Filter;
+import denominator.model.ResourceRecordSet;
+import denominator.model.Zone;
+import denominator.verisignmdns.VrsnMdns.Record;
+
+public class VerisignMDNSAllProfileResourceRecordSetApi implements denominator.AllProfileResourceRecordSetApi {
+    private final String domainName;
+    private final VrsnMdns api;
+    private final Provider<Credentials> credentialsProvider;
+
+    VerisignMDNSAllProfileResourceRecordSetApi(Provider<Credentials> credentialsProvider, String domainName,
+            VrsnMdns api) {
+        this.domainName = domainName;
+        this.api = api;
+        this.credentialsProvider = credentialsProvider;
+    }
+
+    /**
```

Comment:
keep in mind, less code, less things to break, less tests to write, less code to review. These are all good things for the long-term health of denominator and this provider.